x/perp/v2/keeper/hooks.go

```diff
@@ -50,6 +50,12 @@ func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ uint64)
 		// See https://www.notion.so/nibiru/Funding-Payments-5032d0f8ed164096808354296d43e1fa for an explanation of these terms.
 		premiumFraction := markTwap.Sub(indexTwap).QuoInt64(int64(intervalsPerDay))
 
+		if premiumFraction.GT(market.MaxFundingRate) {
+			premiumFraction = market.MaxFundingRate
```

Comment:
I think we're mixing units here. Funding rate (in units of bps) can't be directly added to the premium fraction (units of $/contract). You have to multiply by index price to get premium fraction.

See https://www.notion.so/nibiru/Funding-Payments-5032d0f8ed164096808354296d43e1fa?pvs=4.

Reply:
Yep, but the size needs to be converted to dollars too. For example, if I say the funding rate is 0.01%, then the funding payment is (0.01%)*(size)*(underlying price). We should rename to `funding_rate` and multiply by index price here.

Reply:
It's the first one.

Reply:
You basically just need to clamp Divergence / num_payments_per_day